source/concurrent/Thread.ooc

```diff
@@ -12,19 +12,39 @@ import native/[ThreadUnix, ThreadWin32]
 
 Thread: abstract class {
 	_code: Func
+	_canceled := false
+	_cancelable: Bool
+	_cancelMutex: Mutex = null
+	_ownsCancelMutex: Bool
```

Comment:
Three bools and a mutex seems reasonable to me. Let me get it working and we can evaluate later.